cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -176,9 +176,18 @@ private void setFollowerThrottledReplicas(String topic, Set<String> replicas) {
 
   private void setThrottledReplicas(String topic, Set<String> replicas, boolean throttleLeaderReplica) {
     String replicaThrottleConfigKey = throttleLeaderReplica ? LEADER_THROTTLED_REPLICAS : FOLLOWER_THROTTLED_REPLICAS;
-
-    Properties config = _kafkaZkClient.getEntityConfigs(ConfigType.Topic(), topic);
-    String currThrottledReplicas = config.getProperty(replicaThrottleConfigKey);
+    Properties topicConfigs = _kafkaZkClient.getEntityConfigs(ConfigType.Topic(), topic);
+    if (topicConfigs == null || topicConfigs.isEmpty()) {
+      // Not assume that no topic config means means topic does not exist. Always check topic existence in this situation
+      if (!_kafkaZkClient.topicExists(topic)) {
+        LOG.debug("Skip setting throttled replicas {} for topic {} since topic does not exist", String.join(",", replicas), topic);
+        return;
+      }
+      if (topicConfigs == null) {
+        topicConfigs = new Properties();
+      }
+    }
```

Comment:
1. It's safer to assume `getEntityConfigs` might return `null` since it does not claim it never returns null as its public contract
2. You are right. No need to check the topic existence here.
3. Fixed